src/store/channels/saga.ts

```diff
@@ -45,35 +44,42 @@ export function* markAllMessagesAsRead(channelId, userId) {
 }
 
 // mark all messages as read in current active channel (only if you're not in full screen mode)
-export function* markChannelAsReadIfActive(channelId) {
-  const activeChannelId = yield select(activeChannelIdSelector);
-  if (channelId !== activeChannelId) {
-    return;
-  }
-
+export function* markChannelAsRead(channelId) {
   const currentUser = yield select(currentUserSelector());
   const isMessengerFullScreen = yield select((state) => state.layout.value.isMessengerFullScreen);
-  const activeChannelInfo = activeChannelId ? yield select(rawChannelSelector(activeChannelId)) : null;
+  const channelInfo = yield select(rawChannelSelector(channelId));
 
   // just ensure first that you're not in full screen mode before marking all messages as read in a "channel"
-  if (!isMessengerFullScreen && activeChannelId && activeChannelInfo?.unreadCount > 0) {
-    yield call(markAllMessagesAsRead, activeChannelId, currentUser.id);
+  if (!isMessengerFullScreen && channelInfo?.unreadCount > 0) {
+    yield call(markAllMessagesAsRead, channelId, currentUser.id);
   }
 }
 
 // mark all messages in read in current active conversation
-export function* markConversationAsReadIfActive(channelId) {
-  const activeConversationId = yield select((state) => state.chat.activeConversationId);
-  if (channelId !== activeConversationId) {
+export function* markConversationAsRead(conversationId) {
+  const currentUser = yield select(currentUserSelector());
+  const conversationInfo = yield select(rawChannelSelector(conversationId));
+  if (conversationInfo?.unreadCount > 0) {
+    yield call(markAllMessagesAsRead, conversationId, currentUser.id);
+  }
+}
+
+export function* openChannel(channelId) {
+  if (!channelId) {
     return;
   }
 
-  const currentUser = yield select(currentUserSelector());
-  const activeConversationInfo = activeConversationId ? yield select(rawChannelSelector(activeConversationId)) : null;
+  yield put(setActiveChannelId(channelId));
+  yield call(markChannelAsRead, channelId);
```

Comment:
Just realized, we probably want this to be a `spawn`. There's no reason to delay anything else for this request. Same for the other one.

Reply:
done. thanks